base/int.jl

```diff
@@ -96,6 +96,44 @@ inv(x::Integer) = float(one(x)) / float(x)
 # skip promotion for system integer types
 (/)(x::BitInteger, y::BitInteger) = float(x) / float(y)
 
+
+"""
+    mul_hi(a::T, b::T) where {T<:Base.BitInteger}
+
+Returns the higher half of the product of `a` and `b`.
+
+# Examples
+```jldoctest
+julia> mul_hi(12345678987654321, 123456789)
+82624
+
+julia> (widen(12345678987654321) * 123456789) >> 64
+82624
+
+julia> mul_hi(0xff, 0xff)
```

Comment:
```suggestion
julia> Base.mul_hi(12345678987654321, 123456789)
82624

julia> (widen(12345678987654321) * 123456789) >> 64
82624

julia> Base.mul_hi(0xff, 0xff)
```

Because `mul_hi` isn't exported. This (or `using Base: mul_hi`) is required to make doctests pass.

Reply:
Thanks!